### PR TITLE
Exclude _id from DataStore Dump View

### PR DESCRIPTION
**Pull request by JVickery-TBS**
feat(views): ds dump exclude _id;

- Always exclude the `_id` field from DataStore Dump blueprint.

### CODE REVIEW

File: ckanext/datastore/blueprint.py [hunk 1]
```diff
@@ -44,6 +44,40 @@
...
+    """
+    Always set the list of fields to dump from the DataStore. Excluding to _id column.
+
+    This validator is only used in the dump_schema.
+    """
+    value = data.get(key)
+
+    if not has_request_context() and not hasattr(request, 'view_args') and 'resource_id' not in request.view_args:
+        # treat as ignore_missing
+        data.pop(key, None)
```
Comment (wardi) [inline ckanext/datastore/blueprint.py:61]: `ignore_missing` does
```
if value is missing or value is None:
    data.pop(key, None)
```
Did you mean `treat as ignore`?

File: ckanext/datastore/blueprint.py [hunk 2]
```diff
@@ -44,6 +44,40 @@
...
+        raise StopOnError
+
+    resource_id = request.view_args['resource_id']
+    ds_info = get_action('datastore_info')(context, {'id': resource_id})
+
+    if value is missing or value is None:
```
Comment (wardi) [inline ckanext/datastore/blueprint.py:67]:
```suggestion

    if value is missing or value is None:
        ds_info = get_action('datastore_info')(context, {'id': resource_id})
```

File: ckanext/datastore/blueprint.py [hunk 3]
```diff
@@ -44,6 +44,40 @@
...
+        value = []
+        # _id is never returned from datastore_info
+        for field in ds_info.get('fields', []):
+            value.append(field.get('id'))
```
Comment (wardi) [inline ckanext/datastore/blueprint.py:71]:
```suggestion
        # _id is never returned from datastore_info
        value = [field['id'] for field in ds_info.get('fields', [])]
```

File: ckanext/datastore/blueprint.py [hunk 4]
```diff
@@ -44,6 +44,40 @@
...
+    else:
+        # fields accepts string or list of strings
+        if isinstance(value, text_type):
+            value = value.split(',')
+        if isinstance(value, list):
```
Comment (wardi) [inline ckanext/datastore/blueprint.py:76]:
```suggestion
        if isinstance(value, list) and '_id' in value:
```

File: ckanext/datastore/blueprint.py (hunk 1 above)
Comment (JVickery-TBS) [reply ckanext/datastore/blueprint.py:61]: Comment updated.

**Comment by JVickery-TBS** [comment]
@wardi lemme know if this would be a fine upstream contrib (perhaps added behind a request arg as well such as `?include_id=true|false`).

Also does this need automated tests? Or not really as this is only on the Blueprint level and does not really affect anything else?

**Comment by wardi** [comment]
Might be nice upstream as a `exclude_fields` parameter instead of one specific to `_id`

**Comment by JVickery-TBS** [comment]
@wardi okay I will make an upstream PR next sprint with this

**Comment by JVickery-TBS** [comment]
@wardi is this PR good to go though for our fork?

**Review by wardi — APPROVED** [review]

**Pull request merged**